addons/stock_account/models/stock_move.py

```diff
@@ -437,23 +432,22 @@ def _prepare_move_split_vals(self, uom_qty):
         vals['to_refund'] = self.to_refund
         return vals
 
-    def _create_account_move_line(self, credit_account_id, debit_account_id, journal_id, qty, description, svl_id, cost):
+    def _prepare_account_move_vals(self, credit_account_id, debit_account_id, journal_id, qty, description, svl_id, cost):
         self.ensure_one()
         AccountMove = self.env['account.move'].with_context(default_journal_id=journal_id)
 
         move_lines = self._prepare_account_move_line(qty, cost, credit_account_id, debit_account_id, description)
         if move_lines:
             date = self._context.get('force_period_date', fields.Date.context_today(self))
-            new_account_move = AccountMove.sudo().create({
+            return {
                 'journal_id': journal_id,
                 'line_ids': move_lines,
                 'date': date,
                 'ref': description,
                 'stock_move_id': self.id,
                 'stock_valuation_layer_ids': [(6, None, [svl_id])],
                 'move_type': 'entry',
-            })
-            new_account_move._post()
+            }
 
     def _account_entry_move(self, qty, description, svl_id, cost):
         """ Accounting Valuation Entries """
```

Comment:
Can you remove the useless variables `location_from` and `location_to` of this method ?